tests/e2e/csi_snapshot_basic.go

```diff
@@ -1355,4 +1356,981 @@ var _ = ginkgo.Describe("[block-vanilla-snapshot] Volume Snapshot Basic Test", f
+		snapshot1, err := snapc.SnapshotV1().VolumeSnapshots(newNamespaceName).Create(ctx,
+			getVolumeSnapshotSpec(newNamespaceName, volumeSnapshotClass.Name, pvclaim.Name), metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("Volume snapshot name is : %s", snapshot1.Name)
+
+		defer func() {
+			if !isNamespaceDeleted {
+				if snapshotContentCreated {
+					framework.Logf("Deleting volume snapshot content")
+					err := snapc.SnapshotV1().VolumeSnapshotContents().Delete(ctx,
+						*snapshot1.Status.BoundVolumeSnapshotContentName, metav1.DeleteOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+
+				if snapshotCreated {
+					framework.Logf("Deleting volume snapshot")
+					err := snapc.SnapshotV1().VolumeSnapshots(newNamespaceName).Delete(ctx,
+						snapshot1.Name, metav1.DeleteOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			}
+		}()
+
+		ginkgo.By("Verify volume snapshot is Ready to use")
+		snapshot1_updated, err := waitForVolumeSnapshotReadyToUse(*snapc, ctx, newNamespaceName, snapshot1.Name)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		snapshotCreated = true
+		gomega.Expect(snapshot1_updated.Status.RestoreSize.Cmp(resource.MustParse(diskSize))).To(gomega.BeZero())
+
+		ginkgo.By("Verify volume snapshot content is created")
+		snapshotContent1, err := snapc.SnapshotV1().VolumeSnapshotContents().Get(ctx,
+			*snapshot1_updated.Status.BoundVolumeSnapshotContentName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		snapshotContentCreated = true
+
+		gomega.Expect(*snapshotContent1.Status.ReadyToUse).To(gomega.BeTrue())
+
+		framework.Logf("Get volume snapshot ID from snapshot handle")
+		snapshothandle := *snapshotContent1.Status.SnapshotHandle
+		snapshotId := strings.Split(snapshothandle, "+")[1]
+
+		ginkgo.By("Query CNS and check the volume snapshot entry")
+		err = verifySnapshotIsCreatedInCNS(volHandle, snapshotId)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Delete namespace")
+		err = client.CoreV1().Namespaces().Delete(ctx, newNamespaceName, *metav1.NewDeleteOptions(0))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		isNamespaceDeleted = true
+
+		time.Sleep(1 * time.Minute)
+
+		ginkgo.By("Verify volume snapshot is deleted")
+		_, err = snapc.SnapshotV1().VolumeSnapshots(newNamespaceName).Get(ctx,
+			snapshot1.Name, metav1.GetOptions{})
+		gomega.Expect(err).To(gomega.HaveOccurred())
+
+		content_updated, err := snapc.SnapshotV1().VolumeSnapshotContents().Get(ctx,
+			snapshotContent1.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(content_updated.Name).Should(gomega.Equal(snapshotContent1.Name))
+
+		ginkgo.By("Creating another new namespace for the test")
+		namespace2, err := framework.CreateTestingNS(f.BaseName, client, nil)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		isNamespace2Deleted := false
```

Comment:
no need of this, you can instead add the namespace for default cleanup routines in the framework as follows -  `f.AddNamespacesToDelete(namespace2)`